compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/BIRGenEnv.java

```diff
@@ -62,6 +66,10 @@ class BIRGenEnv {
     Stack<List<BIRBasicBlock>> trapBlocks = new Stack<>();
 
     Map<BlockNode, List<BIRVariableDcl>> varDclsByBlock = new HashMap<>();
+    Map<BVarSymbol, BIRVariableDcl> syntheticVariableDclMap = new ConcurrentHashMap<>();
```

Comment:
Shall we use the `HashMap` here instead of the `ConcurrentHashMap` as discussed in the sync-up meeting